### PR TITLE
admin transaction changes

### PR DESCRIPTION
**Pull request by joshuahannan**
* deposit tokens checks if the account has been registered
* transactions to change parts of the staking contracts
* changing some access modifiers
* making init arguments for the staking contract

### CODE REVIEW

File: contracts/FlowIDTableStaking.cdc [hunk 1]
```diff
@@ -803,8 +803,15 @@ pub contract FlowIDTableStaking {
...
+        pub fun changeMinimumStakeRequirements(_ newRequirements: {UInt8: UFix64}) {
+            pre {
+                newRequirements.keys.length == 5: "Incorrect number of nodes"
+            }
+            FlowIDTableStaking.minimumStakeRequired = newRequirements
+        }
+
         // Changes the total weekly payout to a new value
-        pub fun updateEpochTokenPayout(_ newPayout: UFix64) {
+        pub fun changeEpochTokenPayout(_ newPayout: UFix64) {
```
Comment (robmyers) [inline contracts/FlowIDTableStaking.cdc:814]: I favour personally favour "set" for setters.

File: contracts/FlowIDTableStaking.cdc [hunk 2]
```diff
@@ -970,10 +981,10 @@ pub contract FlowIDTableStaking {
         self.totalTokensStakedByNodeType = {UInt8(1): 0.0, UInt8(2): 0.0, UInt8(3): 0.0, UInt8(4): 0.0, UInt8(5): 0.0}
 
         // 1.25M FLOW paid out in the first week. Decreasing in subsequent weeks
-        self.epochTokenPayout = 1250000.0
+        self.epochTokenPayout = epochTokenPayout
 
-        // initialize the cut of rewards that node operators take to 3%
-        self.nodeDelegatingRewardCut = 0.03
+        // initialize the cut of rewards that node operators take to 10%
```
Comment (robmyers) [inline contracts/FlowIDTableStaking.cdc:986]:
```suggestion
        // initialize the cut of rewards that node operators take
```

File: contracts/FlowIDTableStaking.cdc [hunk 3]
```diff
@@ -970,10 +981,10 @@ pub contract FlowIDTableStaking {
         self.totalTokensStakedByNodeType = {UInt8(1): 0.0, UInt8(2): 0.0, UInt8(3): 0.0, UInt8(4): 0.0, UInt8(5): 0.0}
 
         // 1.25M FLOW paid out in the first week. Decreasing in subsequent weeks
```
Comment (robmyers) [inline contracts/FlowIDTableStaking.cdc:983]:
```suggestion
        // Set initial FLOW paid out in the first week. Decreasing in subsequent weeks
```

**Review by robmyers — APPROVED** [review]

File: contracts/FlowIDTableStaking.cdc (hunk 1 above)
Comment (joshuahannan) [reply contracts/FlowIDTableStaking.cdc:814]: I like that too. I'll change it

**Review by psiemens — APPROVED** [review]

File: contracts/FlowIDTableStaking.cdc [hunk 4]
```diff
@@ -77,7 +77,7 @@ pub contract FlowIDTableStaking {
 
     /// The total amount of tokens that are paid as rewards every epoch
     /// could be manually changed by the admin resource
-    pub var epochTokenPayout: UFix64
+    access(contract) var epochTokenPayout: UFix64
```
Comment (Lefty66) [inline contracts/FlowIDTableStaking.cdc:80]: 57

File: contracts/FlowIDTableStaking.cdc [hunk 5]
```diff
@@ -86,7 +86,7 @@ pub contract FlowIDTableStaking {
 
     /// The percentage of rewards that every node operator takes from 
     /// the users that are delegating to it
-    pub var nodeDelegatingRewardCut: UFix64
+    access(contract) var nodeDelegatingRewardCut: UFix64
```
Comment (Lefty66) [inline contracts/FlowIDTableStaking.cdc:89]: +803

**Pull request merged**